Build fails when profile server-guides is not specified

### Describe the bug

When starting using

```
mvn clean install
```

the build build fails with 

```
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:3.0.0:java (build-web) on project KeycloakWebBuilder: An exception occured while executing the Java class. Multiple compatible overloaded variations were found with the same priority.
[ERROR] The FTL type of the argument values were: Null.
[ERROR] The Java type of the argument values were: Null.
[ERROR] The matching overload was searched among these members:
[ERROR]     org.keycloak.webbuilder.Links.get(org.keycloak.webbuilder.Guides$Guide),
[ERROR]     org.keycloak.webbuilder.Links.get(org.keycloak.webbuilder.Blogs$Blog)
[ERROR] 
[ERROR] ----
[ERROR] FTL stack trace ("~" means nesting-related):
[ERROR]         - Failed at: ${links.get(guides.getGuide("server",...  [in template "templates/documentation-9.ftl" at line 42, column 22]
[ERROR]         - Reached through: #include "../templates/documentation-...  [in template "pages/documentation.ftl" at line 8, column 5]
[ERROR]         ~ Reached through: #nested  [in template "templates/template.ftl" in macro "page" at line 58, column 1]
[ERROR]         ~ Reached through: @tmpl.page current="docs" title="Docu...  [in template "pages/documentation.ftl" at line 3, column 1]
[ERROR] ----
``` 

### Version

17.0.0

### Expected behavior

The build should complete 

### Actual behavior

The build fails.

### How to Reproduce?

Start `mvn clean install`

### Anything else?

The build completes only when the profile is passed: 

```
mvn clean package -Pserver-guides
```

This is due to that `guides.getGuide('server', 'containers')` in 

```
<a href="${links.get(guides.getGuide('server', 'containers'))}">
``` 

is returning null, and therefore the `links.get()` fails as it can't determine the right overloaded method.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -22,7 +22,7 @@ jobs:
     - name: Build with Maven
       env:
         GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
-      run: mvn package -Dpublish -Pserver-guides
+      run: mvn package -Dpublish
     - name: Install SSH Client 🔑
       uses: webfactory/ssh-agent@v0.4.1
       with:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -19,4 +19,4 @@ jobs:
       with:
         java-version: 11
     - name: Build with Maven
-      run: mvn package -Pserver-guides
+      run: mvn package

--- a/README.md
+++ b/README.md
@@ -4,11 +4,7 @@ Can be built locally from IDE by running AutoBuilder main method or with Maven b
 
     mvn clean install
     
-After the build is compeleted open `target/web/index.html` in your favourite web browser.
-    
-To include server guides from `https://github.com/keycloak/keycloak/tree/main/docs/guides` first you need to build those guides an install to your local maven repository, then you can build the website with:
-
-    mvn clean install -Pserver-guides
+After the build is completed open `target/web/index.html` in your favourite web browser.
     
 There's also an auto-builder utility that watches the filesystem and continiously rebuilds the website. This can be very useful when working on the website, especially making stylesheet changes, etc. This can be ran from your IDE by running `AutoBuilder`, or from the command-line with:
 
@@ -20,3 +16,16 @@ To serve the website locally with a web browser specify the base URL with `KC_UR
     mvn install
     
 Finally, when building the website to be published you need to include `-Dpublish`. This should usually not be done manually though as there is a GitHub Action that takes care of building and publishing to `https://github.com/keycloak/keycloak.github.io`.
+
+## Server and Operator guides
+
+Server and Operator guide sources are located at `https://github.com/keycloak/keycloak/tree/main/docs/guides` and are retrieved as a Maven dependency. By default the latest released version is used, but if you want to build the website with the latest guides from `main` first build the guides:
+
+    git clone https://github.com/keycloak/keycloak
+    cd keycloak
+    mvn -pl docs clean install
+
+Then you can build the website with the `999-snapshot` version:
+
+    cd keycloak-web
+    mvn clean install -Dversion.keycloak=999-SNAPSHOT

--- a/pom.xml
+++ b/pom.xml
@@ -176,6 +176,9 @@
     <profiles>
         <profile>
             <id>server-guides</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
             <dependencies>
                 <dependency>
                     <groupId>org.keycloak</groupId>

